### PR TITLE
Upgrade react-router-dom@4.0.0 and other deps

### PR DESCRIPTION
**Pull request by diondirza**
Just quick fix and update dependencies in the project,

### CODE REVIEW

File: shared/components/DemoApp/Error404/index.js [hunk 1]
```diff
@@ -15,10 +15,9 @@ class Error404 extends Component {
   }
 }
 
-Error404.contextTypes = {
-  history: PropTypes.shape({
-    staticContext: PropTypes.object,
-  }).isRequired,
+Error404.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
+  staticContext: PropTypes.object.isRequired,
```
Comment (ctrlplusb) [inline shared/components/DemoApp/Error404/index.js]: Should the `isRequired` be dropped?

Comment (diondirza) [inline shared/components/DemoApp/Error404/index.js]: No, It should not. I prefer not to provide default value for `staticContext` prop.

Comment (ctrlplusb) [inline shared/components/DemoApp/Error404/index.js]: Sweet, lets get this rolled in!

File: internal/webpack/configFactory.js [hunk 2]
```diff
@@ -372,7 +372,7 @@ export default function webpackConfigFactory(buildOptions) {
                 // ES201X code into ES5, safe for browsers.  We exclude module
                 // transilation as webpack takes care of this for us, doing
                 // tree shaking in the process.
-                ifClient(['latest', { es2015: { modules: false } }]),
+                ifClient(['env', { es2015: { modules: false } }]),
```
Comment (ctrlplusb) [inline internal/webpack/configFactory.js]: Please change this back, we have a discussion in #393 for this change

File: package.json [hunk 3]
```diff
@@ -122,38 +122,37 @@
...
     "babel-jest": "19.0.0",
-    "babel-loader": "6.3.2",
+    "babel-loader": "6.4.0",
     "babel-plugin-transform-react-constant-elements": "6.23.0",
     "babel-plugin-transform-react-inline-elements": "6.22.0",
     "babel-plugin-transform-react-jsx-self": "6.22.0",
     "babel-plugin-transform-react-jsx-source": "6.22.0",
     "babel-polyfill": "6.23.0",
-    "babel-preset-env": "1.2.0",
-    "babel-preset-latest": "6.22.0",
```
Comment (ctrlplusb) [inline package.json]: You will need to re-add this dep

Comment (diondirza) [inline package.json]: done in [1159a18](https://github.com/ctrlplusb/react-universally/pull/396/commits/1159a1853162ad9a828358820cad5dd1b171ab92)

**Comment by ctrlplusb** [comment]
Ok, I am going to merge this is, but there is one issue outstanding:

The yarn.lock file has not been updated.  I am assuming you aren't using `yarn`.

I'll do it myself, but I recommend you install `yarn` (it's awesome) and then to upgrade dependencies run:

```
yarn upgrade-interactive --exact
```

You can then check the ones to upgrade.  It will also update the package.json file accordingly. 😀

**Comment by ctrlplusb** [comment]
Done! Thanks @diondirza! 🎉

**Pull request merged**